### PR TITLE
UnitTest  evm/watcher/tx

### PR DESCRIPTION
**Pull request by bylingo**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/watcher/tx_test.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
+package watcher
+
+import (
```
Comment (LeoGuo621) [inline x/evm/watcher/tx_test.go:3]: import should be sorted by “goimports” tool in Goland.

**Comment by MrThingchaiWaon** [comment]
bc1qqaler5c2ulw0mfmvraueyur50u3e22dkkurxcw

File: x/evm/watcher/tx_test.go (hunk 1 above)
Comment (bylingo) [reply x/evm/watcher/tx_test.go:3]: Imports are sorted using goimports in the new commit: https://github.com/okx/exchain/pull/3034/commits/69877dc1fe72636270b4771450c984fe782543ef

**Review by LeoGuo621 — DISMISSED** [review]

**Review by giskook — DISMISSED** [review]

**Comment by bylingo** [comment]
New commit have made following changes:
1. Because many import cycles occur when this code file is included in package watcher, package name is changed into **watcher_test**
2. Some tested functions and some fields of structs are private in package watcher, thus **export_test.go** with some public function is created so as to export these useful functions and fields
3. TestCreateWatchTx is **finished** with no error and all cases in it have passed the test
4. TestRecordTxAndFailedReceipt is still under **debugging**. 
5. For reusing some test functions, some functions with the suffix of WithArg is implemented. They can be further used as the core component of unit test.

**Comment by bylingo** [comment]
New commit finished the unit test of RecordTxAndFailedReceipt. Some important coding ideas are as follow:

1. Because using NewOKExChainApp by app.Setup will cause error when the tx_test.go and watcher_test.go are executed consecutively, tx_test.go chose to initialize a new OKExChainApp by initializing a new WatchTestSt in watcher_test.go and  then inherit the OKExChainApp in it.
2. Because this unit test only aims to test if these branch-statements function properly, the correct execution of this case is only  identified by **the number of batches** added in the watcher.

**Review by giskook — APPROVED** [review]

**Pull request merged**